src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,160 @@
+      args.append(f"--pylint-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    adaptor = PythonTargetAdaptor(
+      sources=EagerFilesetWithSpec(self.source_root, {'globs': []}, snapshot=input_snapshot),
+      address=Address.parse(f"{self.source_root}:target"),
+      compatibility=[interpreter_constraints] if interpreter_constraints else None,
+    )
+    if origin is None:
+      origin = SingleAddress(directory="test", name="target")
+    target = PylintTarget(PythonTargetAdaptorWithOrigin(adaptor, origin))
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_precise_file_args(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+  
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
+  def test_respects_config_file(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], config="[pylint]\ndisable = C0114\n")
+    assert result.exit_code == 0
+    assert result.stdout.strip() == ""
+
+  def test_respects_passthrough_args(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], passthrough_args="--disable=C0114")
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_includes_direct_dependencies(self) -> None:
+    self.create_python_library(name="library")
+    self.create_python_library(
+      name="dependency", sources=["dependency.py"], dependencies=[":library"],
+    )
+    self.write_file(
+      FileContent(
+        path=f"{self.source_root}/dependency.py",
+        content=dedent(
+          """\
+          from nonexistent import erroneous
```

Comment:
Why does this test pass?

Reply:
Pylint requires dependencies in the chroot for it to detect imports on the file(s) being linted; however, it does not care about linting those dependencies. This test validates that Pylint will not lint a direct dependency (lacking docstring and erroneous import). In that vein, it passes because Pylint never executes the erroneous import.

Reply:
Hopefully this is more obvious with the refactored test files 😄